src/Plugin/CookiePlugin.php

```diff
@@ -69,7 +69,7 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
                     }
 
                     // Restrict setting cookie from another domain
-                    if (false === strpos($cookie->getDomain(), $request->getUri()->getHost())) {
+                    if (false === strpos($request->getUri()->getHost(), $cookie->getDomain())) {
```

Comment:
> afaik the server could send the leading . already, we should handle that properly as well, not that we end up with "..symfony.slack.com".

This is already handled by [Http\Message\Cookie::normalizeDomain()](https://github.com/php-http/message/blob/master/src/Cookie.php#L497), so we shouldn't care about it.

> are you sure that cookies should propagate down several levels? like test.com cookie being sent to foo.bar.baz.test.com ?

Yes, that's what is said in RFC 6265. If Domain attribute is not specified, it defaults to the host portion of the current document location (but not including subdomains). If a domain is specified, subdomains are always included.